apps/agreement/contracts/Agreement.sol

```diff
@@ -0,0 +1,1350 @@
+        challenge.actionId = _actionId;
+        challenge.challenger = _challenger;
+        challenge.context = _context;
+        challenge.settlementOffer = _settlementOffer;
+        challenge.endDate = getTimestamp64().add(_requirement.challengeDuration);
+        challenge.challengerFinishedEvidence = _finishedSubmittingEvidence;
+
+        // Transfer challenge collateral
+        _transferFrom(_requirement.token, _challenger, _requirement.challengeAmount);
+
+        // Transfer half of the Arbitrator fees
+        (, , IArbitrator arbitrator) = _getSettingFor(_action);
+        (, ERC20 feeToken, uint256 feeAmount) = arbitrator.getDisputeFees();
+        uint256 arbitratorFees = feeAmount / 2;
+        challenge.arbitratorFeeToken = feeToken;
+        challenge.arbitratorFeeAmount = arbitratorFees;
+        _transferFrom(feeToken, _challenger, arbitratorFees);
+        return challengeId;
+    }
+
+    /**
+    * @dev Dispute an action
+    * @param _action Action instance to be disputed
+    * @param _challenge Current challenge instance for the action
+    * @return _arbitrator Address of the IArbitrator applicable for the action
+    * @return _metadata Metadata content to be used for the dispute
+    * @return Identification number of the dispute created in the arbitrator
+    */
+    function _createDispute(Action storage _action, Challenge storage _challenge, IArbitrator _arbitrator, bytes memory _metadata)
+        internal
+        returns (uint256)
+    {
+        // Compute missing fees for dispute
+        ERC20 challengerFeeToken = _challenge.arbitratorFeeToken;
+        uint256 challengerFeeAmount = _challenge.arbitratorFeeAmount;
+        (address disputeFeeRecipient, ERC20 feeToken, uint256 missingFees, uint256 totalFees) = _getMissingArbitratorFees(
+            _arbitrator,
+            challengerFeeToken,
+            challengerFeeAmount
+        );
+
+        // Pull arbitration fees from submitter
+        address submitter = _action.submitter;
+        _transferFrom(feeToken, submitter, missingFees);
+
+        // Create dispute. The arbitrator should pull any arbitration fees from this Agreement here.
+        // To be safe, We first set the allowance to zero in case there is a remaining approval for the arbitrator.
+        // This is not strictly necessary for ERC20s, but some tokens, e.g. MiniMe (ANT and ANJ),
+        // revert on an approval if an outstanding allowance exists
+        _approveArbitratorFeeTokens(feeToken, disputeFeeRecipient, 0);
+        _approveArbitratorFeeTokens(feeToken, disputeFeeRecipient, totalFees);
+        uint256 disputeId = _arbitrator.createDispute(DISPUTES_POSSIBLE_OUTCOMES, _metadata);
+
+        // Return any remaining portion of the pre-paid arbitrator fees to the challenger, if necessary
+        if (challengerFeeToken != feeToken) {
+            _transfer(challengerFeeToken, _challenge.challenger, challengerFeeAmount);
```

Comment:
⚠️ I think there may be a problem if the arbitrator lowers its fees to below what the challenger pre-paid.

Reply:
Good catch! I'll add a test for this :)